Like button

LovedeepSingh19 opened the pull request:
You can test the like button in marketplace packages.

<img width="1080" alt="Screenshot 2023-12-28 at 12 28 10 PM" src="https://github.com/sugarcane-ai/sugarcane-ai/assets/91473234/06fe3ab0-aaed-490e-a6c7-21addbd8c97f">
--
<img width="1199" alt="Screenshot 2023-12-28 at 12 29 32 PM" src="https://github.com/sugarcane-ai/sugarcane-ai/assets/91473234/87e87234-3122-4021-9f17-75fbcd73f817">

<img width="1360" alt="Screenshot 2024-01-09 at 12 18 30 PM" src="https://github.com/sugarcane-ai/sugarcane-ai/assets/91473234/c822aca1-91ee-4e8a-a8fc-45fd2eb8ded1">

LovedeepSingh19 committed:
Like button

LovedeepSingh19 committed:
minor changes -- like button

devilankur18 commented on apps/factory/prisma/schema.prisma:
```diff
@@ -222,3 +222,28 @@ model PromptLog {
+model Like {
+  id         String     @id @default(uuid())
+  likes      LikeUser[]
+  likesCount Int
```
add default value to 0

devilankur18 commented on apps/factory/src/components/marketplace/like_button.tsx:
```diff
@@ -1,14 +1,76 @@
+  const [LikeId, setLikeId] = useState<string>("");
+
+  useEffect(() => {
+    data?.likes.some((like) => {
+      if (like.userId === sessionData?.user.id) {
+        setHasLiked(true);
+        setLikeId(like.likeId!);
+      }
+    });
```
@LovedeepSingh19 this is not scalable, you can't pull all the likes to frontend. The user id need to send to the backend and check in db. This need to be changed.

LovedeepSingh19 replied:
You told me to put that code in frontend. Previously in meet I showed you the code and 1st line was checking that same thing

devilankur18 commented on apps/factory/src/components/marketplace/like_button.tsx:
```diff
@@ -1,14 +1,76 @@
+    console.log(data?.likesCount);
+    console.log(data?.likes);
+
+    UpdateMutation.mutate(
+      {
+        EntityId,
+        EntityType,
+        hasLiked,
+        LikeId,
+      },
+      {
+        onSuccess() {
+          console.log("Likes Updated");
+        },
+        onError(error) {
+          const errorData = JSON.parse(error.message);
+          console.log("error: ", errorData);
+        },
+      },
+    );
+
+    if (hasLiked) {
+      setCounter(counter! - 1);
+      setHasLiked(!hasLiked);
+    } else {
+      setCounter(counter! + 1);
+      setHasLiked(!hasLiked);
```
We need to discuss this. The hasLiked should not be sent back to the like. Like action should be done and completlely handled with out hasLiked information from frontend.

devilankur18 commented on apps/factory/src/server/api/routers/likes.ts:
```diff
@@ -0,0 +1,110 @@
+
+      return ctx.prisma
+        .$transaction(async (prisma) => {
+          if (hasLiked) {
+            // Delete the existing like from LikeUser table
+            await prisma.likeUser.deleteMany({
+              where: {
+                userId: ctx.session?.user.id,
+                likeId: LikeId,
+              },
+            });
+
```
Create seperate action for unlike and like, don't use the same mutation. We need to enable caching and other mechanism for the same in the near future.

devilankur18 committed:
Docker (#117)

Containerised next application in the pnpm monorepo setup and github actions for build.

devilankur18 committed:
Docker image versioning. (#133)

* fix for docker image tags

LovedeepSingh19 committed:
Like button (Updated)

LovedeepSingh19 committed:
bug fix

Rahul917-Kumar committed:
Cube Sharing (#115)

* add next-seo and share modal

* few correction regarding dummy checkbox

* add react-share and next-seo for cube sharing

* remove .vscode folder, used for testing purpose

* added ISR

* fixes for share url and ssr true

---------

Co-authored-by: Ankur Agarwal <signup.ankur@gmail.com>

Rahul917-Kumar committed:
Cube Sharing (#115)

* add next-seo and share modal

* few correction regarding dummy checkbox

* add react-share and next-seo for cube sharing

* remove .vscode folder, used for testing purpose

* added ISR

* fixes for share url and ssr true

---------

Co-authored-by: Ankur Agarwal <signup.ankur@gmail.com>

Rahul917-Kumar committed:
Cube Sharing (#115)

* add next-seo and share modal

* few correction regarding dummy checkbox

* add react-share and next-seo for cube sharing

* remove .vscode folder, used for testing purpose

* added ISR

* fixes for share url and ssr true

---------

Co-authored-by: Ankur Agarwal <signup.ankur@gmail.com>

devilankur18 committed:
dockerfile fixes

LovedeepSingh19 committed:
fixes 1

devilankur18 committed:
og response default image (#136)

* added og image

* updating path

devilankur18 committed:
fixes for docker

LovedeepSingh19 committed:
Like Added SugarCubes

LovedeepSingh19 committed:
Like Added SugarCubes

LovedeepSingh19 committed:
Like Added SugarCubes

LovedeepSingh19 committed:
Merge branch 'develop' into 125-like-button

LovedeepSingh19 committed:
Update package.json from "develop'
